include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
-        // Get Start/End/Duration from Google Event
-        $starttime = strtotime($event_remote->getStart()['dateTime']);
-        $endtime = strtotime($event_remote->getEnd()['dateTime']);
-        $diff = abs($starttime - $endtime);
-        $tmins = $diff/60;
-        $hours = floor($tmins/60);
-        $mins = $tmins%60;
-
-        // Set Start/End/Duration in SuiteCRM Meeting and Assigned User
-        $event_local->date_start = date("m/d/Y h:i:sa", $starttime);
-        $event_local->date_end = date("m/d/Y h:i:sa", $endtime);
-        $event_local->duration_hours = $hours;
-        $event_local->duration_minutes = $mins;
-        $event_local->assigned_user_id = $this->workingUser->id;
-
-        // Disable all popup reminders for the SuiteCRM meeting
-        $sql = "UPDATE reminders SET popup = '0' WHERE related_event_module_id = '" .
-            $event_local->id .
-            "' AND deleted = '0'";
-        $res = $this->db->query($sql);
-
-        // Mark all reminders where both popup and email are disabled as deleted.
-        $sql = "UPDATE reminders SET deleted = '1' " .
-            "WHERE popup = '0' AND email = '0' AND related_event_module_id = '" .
-            $event_local->id .
-            "' AND deleted = '0'";
-        $res = $this->db->query($sql);
-
-        // Get Google Event Popup Reminders
-        $gReminders = $event_remote->getReminders();
-        $overrides = $gReminders->getOverrides();
-
-        // Create a new popup reminder for each google reminder
-        foreach ($overrides as $override) {
-            if ($override->getMethod() == 'popup') {
-                $sReminder = new Reminder;
-                $sReminder->popup = '1';
-                $sReminder->timer_popup = $override->getMinutes() * 60;
-                $sReminder->related_event_module = $event_local->module_name;
-                $sReminder->related_event_module_id = $event_local->id;
-                $reminderId = $sReminder->save(false);
+        // First, we look for SuiteCRM meetings that are not on Google
+        foreach ($meetings as $meeting) {
+            $gevent = null;
+            if ( !empty($meeting->gsync_id) ) {
+                $gevent = $this->getGoogleEventById($meeting->gsync_id);
+            }
+            
+            $action = $this->pushPullSkip($meeting, $gevent);
+            $actionResult = $this->doAction($action, $meeting, $gevent);
 
-                $reminderInvitee = new Reminder_Invitee;
-                $reminderInvitee->reminder_id = $reminderId;
-                $reminderInvitee->related_invitee_module = 'Users';
-                $reminderInvitee->related_invitee_module_id = $this->workingUser->id;
-                $reminderInvitee->save(false);
+            if (!$actionResult) {
+                // TODO: confusing info log: $actionResult evaluated as string but it also false if converted to boolean. There is an empty string in log as result. - TODO: Inform the caller about this problem (exception?)
```

Comment:
@gymad 
Yep. Very confusing. I think this is left over cruft. doAction() was changed significantly. It's never going to return a string, only true or throw exception.

Reply:
Removed if statement